processor/filterprocessor/filter_processor_logs_test.go

```diff
@@ -393,16 +393,16 @@ func testResourceLogs(lwrs []logWithResource) plog.Logs {
 	for i, lwr := range lwrs {
 		rl := ld.ResourceLogs().AppendEmpty()
 
-		// Add resource level attribtues
+		// Add resource level attributes
 		pcommon.NewMapFromRaw(lwr.resourceAttributes).CopyTo(rl.Resource().Attributes())
 		ls := rl.ScopeLogs().AppendEmpty().LogRecords()
 		for _, name := range lwr.logNames {
 			l := ls.AppendEmpty()
-			l.Attributes().InsertString("name", name)
-			// Add record level attribtues
+			// Add record level attributes
 			for k := 0; k < ls.Len(); k++ {
 				pcommon.NewMapFromRaw(lwrs[i].recordAttributes).CopyTo(ls.At(k).Attributes())
 			}
+			l.Attributes().InsertString("name", name)
```

Comment:
If tests were passing before with "name" being dropped, is this attribute necessary?  Or do the tests need more changes to actually check that the "name" attribute has the correct value?

Reply:
It verifies which log record has been passed through filtering process.

> If tests were passing before with "name" being dropped, is this attribute necessary? 
Tests are valid, so there is no more changes required.

It is related to this kind of code:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/65f2b80b0a89fb5a0c5c0da15a499b5dcaf9e66a/processor/filterprocessor/filter_processor_logs_test.go#L235-L238

Now, only count is compared, after fix, names also will be verified

Name verification is optional and is skipped if `name` is not set:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/65f2b80b0a89fb5a0c5c0da15a499b5dcaf9e66a/processor/filterprocessor/filter_processor_logs_test.go#L378-L382

Reply:
I fixed it, now we are adding record attributes only to newly added record (without iterating previous ones)